datadog/dogstatsd/base.py

```diff
@@ -511,6 +505,15 @@ def socket(self, new_socket):
         if new_socket:
             try:
                 self._socket_kind = new_socket.getsockopt(socket.SOL_SOCKET, socket.SO_TYPE)
+                if self._socket_kind == socket.SOCK_STREAM:
+                    self._transport = "uds-stream"
+                    self._max_payload_size = self._max_buffer_len or UDS_OPTIMAL_PAYLOAD_LENGTH
+                elif self._socket_kind == socket.SOCK_DGRAM:
+                    self._transport = "uds"
+                    self._max_payload_size = self._max_buffer_len or UDS_OPTIMAL_PAYLOAD_LENGTH
+                else:
```

Comment:
Aren't udp sockets `SOCK_DGRAM` too?

Reply:
I've updated the way I check for they type of socket.